test/functional/tests/initialize/test_startup_init_config.py

```diff
@@ -102,3 +102,76 @@ def test_cas_startup(cache_mode, filesystem):
         core.unmount()
         InitConfig.create_default_init_config()
         casadm.stop_all_caches()
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_mode_pair", [(CacheMode.WT, CacheMode.WB),
+                                              (CacheMode.WB, CacheMode.WA),
+                                              (CacheMode.WA, CacheMode.PT),
+                                              (CacheMode.PT, CacheMode.WO),
+                                              (CacheMode.WO, CacheMode.WT)])
+def test_cas_start_with_changed_mode(cache_mode_pair):
+    """
+    title: Check starting cache in other cache mode by starting OpenCAS service from config.
+    description: |
+      Start cache, create config based on running configuration but with another cache mode,
+      restart OpenCAS service and check if cache defined in config file starts properly.
+      Check all cache modes.
+    pass_criteria:
+      - Cache starts with attached core
+      - Cache starts in mode saved in configuration file.
+    """
+    with TestRun.step("Prepare partitions for cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(200, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(400, Unit.MebiByte)])
+        core_part = core_dev.partitions[0]
+
+    with TestRun.step(f"Start cache in the {cache_mode_pair[0]} mode and add core."):
+        cache = casadm.start_cache(cache_part, cache_mode_pair[0], force=True)
+        core = cache.add_core(core_part)
+
+    with TestRun.step(
+            f"Create the configuration file with a different cache mode ({cache_mode_pair[1]})"
+    ):
+        init_conf = InitConfig()
+        init_conf.add_cache(cache.cache_id, cache.cache_device, cache_mode_pair[1])
+        init_conf.add_core(cache.cache_id, core.core_id, core.core_device)
+        init_conf.save_config_file()
+
+    with TestRun.step("Re-init and restart OpenCAS service."):
+        casadm.stop_all_caches()
+        casctl.init(True)
+        casctl.start()
+
+    with TestRun.step("Check if cache started in correct mode with core attached."):
+        validate_cache(cache, cache_mode_pair[1])
+
+
+def validate_cache(cache, cache_mode):
+    try:
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            raise Exception
+    except Exception:
+        TestRun.fail(
+            f"Cache did not load successfully - wrong number of caches: {caches_count}."
```

Comment:
if len() method will raise exception due to none output of  casadm_parser.get_caches(), than caches_count will be not defined and will raise exception in this line.
I suggest executing casadm_parser.get_caches() in the begining of this method and use only it's output later to check if number of caches is OK and also to check if cache mode of returned cache is OK - then you can remove cache parameter from this method.